src/passes/OptimizeInstructions.cpp

```diff
@@ -189,6 +190,15 @@ struct OptimizeInstructions
     }
   }
 
+  EffectAnalyzer effects(Expression* expr) {
+    return EffectAnalyzer(getPassOptions(), getModule()->features, expr);
+  }
+
+  bool commutable(Expression* a, Expression* b) {
```

Comment:
If we think `commutable` is a better name than `canReorder`, let's add a TODO to rename the latter? (as it seems odd to have it call a method with a different name here)

However I'm not sure I prefer it, as `commutable` may have a wider meaning in general, can't it? (not just ordering)

Reply:
I guess normally an operation is commutable, but here I am describing the operands as commutable. I'll rename this to `canReorder`.

Reply:
Oh nice, it's even the same number of characters 😄